homeassistant/components/camera/xiaomi.py

```diff
@@ -86,55 +129,105 @@ def model(self):
         return self._model
 
     def get_latest_video_url(self):
-        """Retrieve the latest video file from the Xiaomi Camera FTP server."""
+        """
+        Retrieve the latest video file from the Xiaomi Camera FTP/HTTP server.
+        """
         from ftplib import FTP, error_perm
+        import requests
+
+        if self.protocol == 'http':
+            if self.model == MODEL_MJ1080:
+                # Get last dir
+                try:
+                    res = requests.get('http://{0}:{1}{2}'.format(self.host,
+                                                                  self.port,
+                                                                  self.path))
+                except Exception as e:
+                    _LOGGER.error("Error when making request: %r" % e)
+                    return None
+                parser = Mj1080HtmlParser()
+                parser.feed(res.text)
+                try:
+                    last_dir = parser.get_contents()[-1]
+                except IndexError:
+                    _LOGGER.warning("There don't appear to be any uploaded "
+                                    "videos (no folder found)")
+                    return None
+
+                # Get second last MP4
+                try:
+                    res = requests.get('http://{0}:{1}{2}/{3}'.format(
```

Comment:
The device interface, I/O and data parsing, should be extracted to standalone library published on pypi.